src/benchmarks/micro/coreclr/perflab/DictinoaryExpansion.cs

```diff
@@ -0,0 +1,158 @@
+                case 4: return typeof(List<List<List<List<T>>>>);
+                case 5: return typeof(List<List<List<List<List<T>>>>>);
+                case 6: return typeof(List<List<List<List<List<List<T>>>>>>);
+                case 7: return typeof(List<List<List<List<List<List<List<T>>>>>>>);
+                case 8: return typeof(List<List<List<List<List<List<List<List<T>>>>>>>>);
+                case 9: return typeof(List<List<List<List<List<List<List<List<List<T>>>>>>>>>);
+                case 10: return typeof(List<List<List<List<List<List<List<List<List<List<T>>>>>>>>>>);
+                case 11: return typeof(List<List<List<List<List<List<List<List<List<List<List<T>>>>>>>>>>>);
+                case 12: return typeof(List<List<List<List<List<List<List<List<List<List<List<List<T>>>>>>>>>>>>);
+                default: return typeof(List<List<List<List<List<List<List<List<List<List<List<List<List<T>>>>>>>>>>>>>);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static Type GenFunc2<T>(int level)
+        {
+            switch (level)
+            {
+                case 0: return typeof(T);
+                case 1: return typeof(List<T>);
+                case 2: return typeof(List<List<T>>);
+                case 3: return typeof(List<List<List<T>>>);
+                case 4: return typeof(List<List<List<List<T>>>>);
+                case 5: return typeof(List<List<List<List<List<T>>>>>);
+                case 6: return typeof(List<List<List<List<List<List<T>>>>>>);
+                case 7: return typeof(List<List<List<List<List<List<List<T>>>>>>>);
+                case 8: return typeof(List<List<List<List<List<List<List<List<T>>>>>>>>);
+                case 9: return typeof(List<List<List<List<List<List<List<List<List<T>>>>>>>>>);
+                case 10: return typeof(List<List<List<List<List<List<List<List<List<List<T>>>>>>>>>>);
+                case 11: return typeof(List<List<List<List<List<List<List<List<List<List<List<T>>>>>>>>>>>);
+                case 12: return typeof(List<List<List<List<List<List<List<List<List<List<List<List<T>>>>>>>>>>>>);
+                default: return typeof(List<List<List<List<List<List<List<List<List<List<List<List<List<T>>>>>>>>>>>>>);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void GenFuncTest()
+        {
+            for (int j = 0; j < s_Iterations; j++)
+            {
+                for (int i = 0; i < 15; i++)
+                    GenFunc<string>(i);
+
+                for (int i = 0; i < 15; i++)
+                    GenFunc<object>(i);
+
+                for (int i = 0; i < 15; i++)
+                    GenFunc2<object>(i);
+
+                for (int i = 0; i < 15; i++)
+                    GenFunc<DictionaryExpansion>(i);
+            }
+        }
+
+        public static int s_Iterations = 100000;
+
+        [Benchmark]
```

Comment:
would it be possible to add some comment here explaining what the benchmark is measuring? A link to some coreclr docs would be enough